### PR TITLE
Fix CUDA 11.6 build error on Windows

### PR DESCRIPTION
**Pull request by skottmckay**
**Description**: 
A Windows SDK header #defines `small`, which clashes with a typedef in the CUDA 11.6 (and possibly other versions) headers.

Add an indirection for including cub.cuh that undefines `small`.

**Motivation and Context**
#11227

### CODE REVIEW

File: onnxruntime/contrib_ops/cuda/bert/longformer_attention_impl.h [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 #pragma once
 #include "core/providers/cuda/shared_inc/cuda_utils.h"
+#include <cublas_v2.h>
```
Comment (yuslepukhin) [inline onnxruntime/contrib_ops/cuda/bert/longformer_attention_impl.h:6]: >include <cublas_v2.h> [](http://example.com/codeflow?start=1&length=21)

This header gets included in both .cc and .cu file. Is it possible to clash with small #define there?

File: onnxruntime/contrib_ops/cuda/bert/longformer_attention_softmax.h [hunk 2]
```diff
@@ -16,6 +16,7 @@ limitations under the License.
 
 // This is fast cuda kernels for longformer attention softmax.
 // It uses two temporary matrix of BxNxSxS, and consumes more memory when sequence length is large.
+#include <cublas_v2.h>
```
Comment (yuslepukhin) [inline onnxruntime/contrib_ops/cuda/bert/longformer_attention_softmax.h:19]: ><cublas_v2.h> [](http://example.com/codeflow?start=9&length=13)

Also gets included into .cc

**Comment by snnn** [comment]
@[JackBoosY](https://github.com/microsoft/onnxruntime/commits?author=JackBoosY) , vcpkg uses CUDA 11.6, how did you make it work?

File: onnxruntime/contrib_ops/cuda/bert/longformer_attention_impl.h (hunk 1 above)
Comment (skottmckay) [reply onnxruntime/contrib_ops/cuda/bert/longformer_attention_impl.h:6]: Not sure what you mean.

I moved it here as this file uses cudaStream_t and cublasHandle_t which are defined via cublas_v2.h.

What #define potentially clashes?

Comment (yuslepukhin) [reply onnxruntime/contrib_ops/cuda/bert/longformer_attention_impl.h:6]: Ok

**Comment by cloudhan** [comment]
Thoes `#defines small` in windows header can be avoided by adding a `WIN32_LEAN_AND_MEAN` define, IIRC.

From the [comment](https://github.com/microsoft/onnxruntime/issues/11227#issuecomment-1100028576), the define is indeed missing from the command.

**Review by hariharans29 — APPROVED** [review]

**Comment by skottmckay** [comment]
Closing in preference for the simpler approach in #11569 11578

**Pull request closed without merging**